kuksa_databroker/databroker/src/glob.rs

```diff
@@ -51,3 +52,64 @@ pub fn to_regex(glob: &str) -> Result<Regex, Error> {
     let re = to_regex_string(glob);
     Regex::new(&re).map_err(|_err| Error::RegexError)
 }
+
+pub fn matches_path_pattern(input: &str) -> bool {
+    if !input.starts_with('*') {
+        let pattern = r"^(?:\w+(?:\.\w+)*)(?:\.\*+)?$";
+        let regex = Regex::new(pattern).unwrap();
+    
+        let pattern2 = r"^[^.]+(\.[^.]+)*\.(\*{2}|\*|[A-Za-z]+)(\.[^.]+)*\.[^.]+$";
+        let regex2 = Regex::new(pattern2).unwrap();
+
+        regex.is_match(input) || regex2.is_match(input)
+    } else {
+        false
+    }
+}
+
+#[tokio::test]
+async fn test_valid_request_path() {
+    let inputs = vec![
```

Comment:
Since the input is split from the assertions below, it would be easier to see which paths are supposed to match by including the "expectation" in the same place.

I.e. something like:
```rust
let inputs = &[
    ("String.*", true),
    ...
];
```